sdk/appplatform/mgmt/src/main/java/com/azure/resourcemanager/appplatform/models/SpringAppDomain.java

```diff
@@ -0,0 +1,16 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.resourcemanager.appplatform.models;
+
+import com.azure.resourcemanager.appplatform.fluent.inner.CustomDomainResourceInner;
+import com.azure.resourcemanager.resources.fluentcore.arm.models.ExternalChildResource;
+import com.azure.resourcemanager.resources.fluentcore.model.HasInner;
+
+/** An immutable client-side representation of an Azure Spring App Custom Domain. */
+public interface SpringAppDomain
+    extends ExternalChildResource<SpringAppDomain, SpringApp>,
+    HasInner<CustomDomainResourceInner> {
+    /** @return the properties of the spring app custom domain */
+    CustomDomainProperties properties();
```

Comment:
Not a clear preference. But it might be better to flatten the `properties` and return its field in this class.

Reply:
I think it is not important interface, so remains the `properties`. Firstly I would even like to return the inner type from get and list.